platform/spark/modules/mdns/Buffer.h

```diff
@@ -0,0 +1,42 @@
+#include "application.h"
+
+#ifndef _INCL_BUFFER
+#define _INCL_BUFFER
+
+#define INVALID_MARK_OFFSET 0xffff
+
+class Buffer {
+    public:
+        Buffer(uint16_t size);
+
+        uint16_t available();
+
+        void mark();
+        void reset();
+        void setOffset(uint16_t offset);
+        uint16_t getOffset();
+
+        void read(UDP * udp);
+
+        uint8_t readUInt8();
+        uint16_t readUInt16();
+
+        void write(UDP * udp);
+
+        void writeUInt8(uint8_t value);
```

Comment:
is this needed? 
Can't the compiler figure it out by the function signature?
write(uint8_t value);
write(uint16_t value);
write(uint32_t value);